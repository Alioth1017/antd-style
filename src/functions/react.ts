import _styled from '@emotion/styled';

import type { CreateStyled } from '@/types';

export const styled = _styled as CreateStyled;
export { css as reactCss, ThemeProvider as EmotionThemeProvider, withTheme } from '@emotion/react';
