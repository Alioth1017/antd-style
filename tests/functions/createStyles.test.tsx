import { render } from '@testing-library/react';
import { createStyles, css } from 'antd-style';

describe('createStyles', () => {
  it('插入样式', async () => {
    const useStyles = createStyles(
      ({ token }) =>
        css`
          color: ${token.colorPrimary};
        `,
    );
    const App = () => {
      const className = useStyles();
      return <div className={className}>123</div>;
    };

    const { container } = render(<App />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
