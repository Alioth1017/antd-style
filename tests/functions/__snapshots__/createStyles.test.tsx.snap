// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStyles styles 对象的使用 createStyleFn 通过函数方式可以拿到 token 等信息 字符串模板的对象模式用法 1`] = `
.emotion-0 {
  background-color: #f5f5f5;
  padding: 24px;
}

.emotion-1 {
  margin-top: 24px;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    card
  </div>
</div>
`;

exports[`createStyles styles 对象的使用 styleObject 方法 对象模式的用法 1`] = `
.emotion-0 {
  background-color: #f5f5f5;
  padding: 24px;
}

.emotion-1 {
  margin-top: 16px;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    card
  </div>
</div>
`;

exports[`createStyles theme 对象使用 需要包裹 ThemeProvider 才能拿到 theme 对象 1`] = `
.emotion-0 {
  color: #1677ff;
}

<div
  class="emotion-0"
  style="background: rgb(245, 245, 245);"
>
  123
</div>
`;
